cloudinit/sources/DataSourceCloudStack.py

```diff
@@ -108,7 +110,12 @@ def _get_domainname(self):
             "Falling back to ISC dhclient"
         )
 
-        lease_file = dhcp.IscDhclient.get_latest_lease()
+        lease_file = None
+        if self.distro:
```

Comment:
please get rid of this if, I think it's misleading.
this reads as: this could sensibly happen! rather than: if this happens, nothing here should work

Reply:
This method was added in my previous PR: https://github.com/canonical/cloud-init/pull/4433
You can follow the discussion there (its quite long, though). 

In short: the virtual machine FQDN is assembled by concatenating isntance name + '.' + domain_name_given_by_dhcp.

This method is only used at get_hostname() method that tries to make an FQDN and if it can't obtain DHCP domain name for any reason, it uses the instance name as FQDN (the behavior previous to my PR)

Reply:
Oh please disregard my previous comment, I thought you commented similar code in get_vr_address() method.

Well, I of course can let it crash here. But what to do with the get_hostname() method that uses this? The current behavior is to try assembling an FQDN and if that fails, return the instance name as FQDN.

Reply:
Updated the PR: removed the distro=None checks, as suggested.